### PR TITLE
FIX: Prepare for PyVista 0.30.0 (part 2)

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR brings the second wave of changes to adapt to PyVista 0.30.0 and PyVistaQt 0.3.0+

ToDo:
- [x] Revert Azure, Circle, `requirements.txt` and `environment.yml` to `pyvista/master` once https://github.com/pyvista/pyvistaqt/pull/95 is merged

Follows https://github.com/mne-tools/mne-python/pull/9274
Related to https://github.com/pyvista/pyvista/pull/1257

### CODE REVIEW

File: mne/viz/backends/_pyvista.py [hunk 1]
```diff
@@ -668,18 +668,13 @@ def _update_picking_callback(self,
...
-            on_button_press
-        )
-        self.plotter.iren.AddObserver(
-            vtk.vtkCommand.EndInteractionEvent,
-            on_button_release
-        )
+        try:
+            add_obs = self.plotter.iren.AddObserver
+        except AttributeError:
+            add_obs = self.plotter.iren.add_observer
```
Comment (hoechenberger) [inline mne/viz/backends/_pyvista.py:674]: I think it would be good to add a comment why we do this / which line corresponds to which version

Comment (GuillaumeFavelier) [reply mne/viz/backends/_pyvista.py:674]: Very good advice, thank you

File: mne/viz/_brain/_brain.py [hunk 2]
```diff
@@ -1692,8 +1692,10 @@ def _clear_callbacks(self):
         # Remove the default key binding
         if getattr(self, "iren", None) is not None:
             try:
+                # pyvista<=0.30.0
                 self.plotter._key_press_event_callbacks.clear()
             except AttributeError:
+                # pyvista>=0.30.0
```
Comment (hoechenberger) [inline mne/viz/_brain/_brain.py:1698]: `== 0.30.0` will match both though?? 😅

Comment (GuillaumeFavelier) [reply mne/viz/_brain/_brain.py:1698]: Whoops

Comment (GuillaumeFavelier) [reply mne/viz/_brain/_brain.py:1698]: It should be fixed in 6f98ca81ce618495e6bf1a3673cf637f64d0d45b

**Comment by GuillaumeFavelier** [comment]
I think Circle does not want to install PyVistaQt master because `dev0` is missing in the version number.

**Comment by GuillaumeFavelier** [comment]
I opened https://github.com/pyvista/pyvistaqt/pull/97#issuecomment-821192485

**Comment by GuillaumeFavelier** [comment]
Okay, with this we should (hopefully) be better prepared for `pyvista 0.30.0`.

File: mne/viz/_brain/_brain.py (hunk 2 above)
Comment (larsoner) [reply mne/viz/_brain/_brain.py:1698]: should probably be `0.3.0` not `0.30.0`

**Comment by larsoner** [comment]
Pushed a tiny cosmit, thanks @GuillaumeFavelier !

**Pull request merged**